packages/SwingSet/test/test-state.js

```diff
@@ -71,187 +70,67 @@ async function testStorage(t, s, getState, commit) {
-  );
-});
-
-test('crankBuffer can abortCrank', t => {
-  const store = initSwingStore(null);
-  const getState = () => getAllState(store).kvStuff;
-
-  const {
-    crankBuffer: s,
-    commitCrank,
-    abortCrank,
-  } = buildCrankBuffer(store.kvStore, createSHA256);
-
-  s.set('foo', 'f');
-  t.truthy(s.has('foo'));
-  t.is(s.get('foo'), 'f');
-
-  s.set('foo2', 'f2');
-  s.set('foo1', 'f1');
-  s.set('foo3', 'f3');
-  t.deepEqual(Array.from(s.getKeys('foo1', 'foo3')), ['foo1', 'foo2']);
-  t.deepEqual(Array.from(s.getKeys('foo1', 'foo4')), ['foo1', 'foo2', 'foo3']);
-
-  s.delete('foo2');
-  t.falsy(s.has('foo2'));
-  t.is(s.get('foo2'), undefined);
-  t.deepEqual(Array.from(s.getKeys('foo1', 'foo4')), ['foo1', 'foo3']);
-
-  checkState(t, getState, []);
-
-  commitCrank();
-  checkState(t, getState, [
-    ['foo', 'f'],
-    ['foo1', 'f1'],
-    ['foo3', 'f3'],
-  ]);
-
-  s.set('foo4', 'f4');
-  abortCrank();
-
-  checkState(t, getState, [
-    ['foo', 'f'],
-    ['foo1', 'f1'],
-    ['foo3', 'f3'],
-  ]);
-
-  s.set('foo5', 'f5');
-  commitCrank();
-  checkState(t, getState, [
-    ['foo', 'f'],
-    ['foo1', 'f1'],
-    ['foo3', 'f3'],
-    ['foo5', 'f5'],
-  ]);
-});
-
 test('storage helpers', t => {
-  const store = initSwingStore(null);
-  const s = addHelpers(store.kvStore);
+  const kernelStorage = initSwingStore(null).kernelStorage;
+  const kv = kernelStorage.kvStore;
 
-  s.set('foo.0', 'f0');
-  s.set('foo.1', 'f1');
-  s.set('foo.2', 'f2');
-  s.set('foo.3', 'f3');
+  kv.set('foo.0', 'f0');
+  kv.set('foo.1', 'f1');
+  kv.set('foo.2', 'f2');
+  kv.set('foo.3', 'f3');
   // omit foo.4
-  s.set('foo.5', 'f5');
-  checkState(t, () => getAllState(store).kvStuff, [
+  kv.set('foo.5', 'f5');
+  checkState(t, () => getAllState(kernelStorage).kvStuff, [
     ['foo.0', 'f0'],
     ['foo.1', 'f1'],
     ['foo.2', 'f2'],
     ['foo.3', 'f3'],
     ['foo.5', 'f5'],
   ]);
 
-  t.deepEqual(Array.from(s.enumeratePrefixedKeys('foo.', 0)), [
-    'foo.0',
-    'foo.1',
-    'foo.2',
-    'foo.3',
-  ]);
-  t.deepEqual(Array.from(s.enumeratePrefixedKeys('foo.', 1)), [
-    'foo.1',
-    'foo.2',
-    'foo.3',
-  ]);
-  t.deepEqual(Array.from(s.getPrefixedValues('foo.', 0)), [
+  t.deepEqual(Array.from(getPrefixedValues(kv, 'foo.')), [
     'f0',
     'f1',
     'f2',
     'f3',
   ]);
-  t.deepEqual(Array.from(s.getPrefixedValues('foo.', 1)), ['f1', 'f2', 'f3']);
-
-  s.deletePrefixedKeys('foo.', 1);
-  t.truthy(s.has('foo.0'));
-  t.falsy(s.has('foo.1'));
-  t.falsy(s.has('foo.2'));
-  t.falsy(s.has('foo.3'));
-  t.falsy(s.has('foo.4'));
-  t.truthy(s.has('foo.5'));
-  checkState(t, () => getAllState(store).kvStuff, [
-    ['foo.0', 'f0'],
-    ['foo.5', 'f5'],
-  ]);
+
+  deletePrefixedKeys(kv, 'foo.');
+  t.falsy(kv.has('foo.0'));
+  t.falsy(kv.has('foo.1'));
+  t.falsy(kv.has('foo.2'));
+  t.falsy(kv.has('foo.3'));
+  t.falsy(kv.has('foo.4'));
+  t.truthy(kv.has('foo.5'));
```

Comment:
let's add a comment just before the `foo.5` line, because this is testing something weird and I had to read it twice before I remembered

```js
// `getPrefixedValues` and `deletePrefixedKeys` both happen to work by counting from zero,
// so if there are any gaps (e.g. 'foo.4'), they'll stop counting
```

I think the `start=0` argument, which you correctly deleted, was a leftover from some ancient time. It's weird for a `deletePrefixedKeys` to react to a gap like this, and means we couldn't use it to delete, e.g., a vat's input queue (which, when we implement it, will be a bunch of keys that don't start at zero). But we'll implement those queues with something better anyways, so I'm happy to see that feature go.